uta_tools/data_sources/mane_transcript.py

```diff
@@ -0,0 +1,634 @@
+"""Module for retrieving MANE Transcript from variation on p/c/g coordinate.
+Steps:
+1. Map annotation layer to genome
+2. Liftover to preferred genome
+    We want to liftover to GRCh38. We do not support getting MANE transcripts
+    for GRCh36 and earlier assemblies.
+3. Select preferred compatible annotation
+4. Map back to correct annotation layer
+"""
+from uta_tools import logger
+from typing import Optional, Tuple, Dict
+import hgvs.parser
+import math
+
+
+class MANETranscript:
+    """Class for retrieving MANE transcripts."""
+
+    def __init__(self, seqrepo_access, transcript_mappings,
+                 mane_transcript_mappings, uta) -> None:
+        """Initialize the MANETranscript class.
+        :param SeqRepoAccess seqrepo_access: Access to seqrepo queries
+        :param TranscriptMappings transcript_mappings: Access to transcript
+            accession mappings and conversions
+        :param MANETranscriptMappings mane_transcript_mappings: Access to
+            MANE Transcript accession mapping data
+        :param UTA uta: UTA instance to give access to query methods for
+            transcript alignments
+        """
+        self.seqrepo_access = seqrepo_access
+        self.hgvs_parser = hgvs.parser.Parser()
+        self.transcript_mappings = transcript_mappings
+        self.mane_transcript_mappings = mane_transcript_mappings
+        self.uta = uta
+
+    def _get_reading_frame(self, pos) -> int:
+        """Return reading frame number.
+        Only used on c. coordinate
+        :param int pos: Position
+        :return: Reading frame
+        """
+        pos_mod_3 = pos % 3
+        if pos_mod_3 == 0:
+            pos_mod_3 = 3
+        return pos_mod_3
```

Comment:
you might be able to drop this as a function and just do something like this wherever you're calling it

```
reading_frame = pos % 3 or 3
```